### PR TITLE
Improve error message when provisioning local kubeflow resources with a non-local container registry.

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
Improve error message when provisioning local kubeflow resources with a non-local container registry.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by htahir1 — COMMENTED** [review]
Wait, I dont really see the error something like "For a remote stack, you must provision the stack yourself. Please follow the guide in the docs to get set up" or something

**Comment by schustmi** [comment]
I implemented the reverse order essentially: Instead of telling them that `zenml stack up` isn't possible with non-local components, the error message tells users to change to a local container registry if they want to provision the local kubeflow deployment (the kubeflow orchestrator itself is agnostic to local/remote settings).

File: src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #  permissions and limitations under the License.
 
 import os
+import re
```
Comment (strickvl) [inline src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py:16]: Why was this added in? I can't see it being used anywhere.

Comment (htahir1) [reply src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py:16]: Its used in line 380

**Review by htahir1 — APPROVED** [review]

**Comment by schustmi** [comment]
@strickvl Is this good to merge in your opinion?

**Review by strickvl — APPROVED** [review]

**Pull request merged**